bazel/utils/binary.bzl

```diff
@@ -0,0 +1,72 @@
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_file")
+
+def _download_binary(ctx):
+    """Downloads a single binary that is not tarballed.
+
+    Example:
+          download_binary(
+                name = "jq_macos_amd64",
+                binary_name = "jq",
+                uri = "https://github.com/stedolan/jq/releases/download/jq-1.6/jq-osx-amd64",
+            )
+    """
+    ctx.download(
+        url = [
+            ctx.attr.uri,
+        ],
+        output = ctx.attr.binary_name,
+        executable = True,
+    )
+    ctx.file(
+        "BUILD.bazel",
+        content = 'exports_files(glob(["*"]))',
+    )
+
+download_binary = repository_rule(
+    _download_binary,
+    attrs = {
+        "binary_name": attr.string(
+            mandatory = True,
+        ),
+        "uri": attr.string(
+            mandatory = True,
+        ),
+    },
+)
+
+def _declare_binary(ctx):
+    """Declares a single binary, used as a wrapper around a select() statement
```

Comment:
move this docstring to be the `doc = """` parameter of `rule(...` (see kernel/defs.bzl for example).

Tl;Dr: documentation generally goes on the rule definition, the function _declare_binary generally gets no docstring, or a simple docstring like """Implements the declare_binary rule."""